compiler+runtime/test/bash/main-exit/pass-test

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bb
+
+(ns jank-test-main-exit
+  (:require [babashka.process :as proc]
+            [clojure.string :as str]
+            [clojure.test :as t :refer [deftest run-test is]]))
+
+(def this-ns (ns-name *ns*))
+
+(deftest jank-run-main-successful-exit
+  (let [{:keys [err out exit]}
+        (proc/shell {:out :string :err :string :continue true}
+                    "jank --module-path src run-main jank-test.successful-exit-code")]
+    (is (empty? err))
+    (is (= "Bottom of clojure.core\njank-test.successful-exit-code: some expected output\n"
```

Comment:
These tests are incredibly fragile, since they're based on the raw stdout of jank. So, not only will they all break when we remove the `(println "Bottom of clojure.core")`, which should not break any tests, but they'll also not work on the error-reporting branch since we're outputting source snippets, stack traces, etc. If this is a test file for exit codes, let's focus on exit codes.

Reply:
Thanks, replaced with:

```clojure
(deftest jank-exit-code-test
  (is-shell-exit 0 "jank --module-path src run-main jank-test.successful-exit-code")
  (is-shell-exit 1 "jank --module-path src run-main jank-test.fail-throw-exit")
  (is-shell-exit 1 "jank --module-path src run-main jank-test.fail-uncatchable-exit")
  (is-shell-exit 106 "jank")
  (is-shell-exit 1 "jank run src/jank_test/throwing_script.jank")
  (is-shell-exit 1 "jank run src/jank_test/uncatchable_script.jank")
  (is-shell-exit 0 "jank run src/jank_test/successful_script.jank"))
```